### PR TITLE
docs: introduce user docs

### PR DESCRIPTION
**Pull request by varl**

### CODE REVIEW

**Comment by varl** [comment]
This publishes the user facing docs at the root URL and the demo site under `/demo`:

- docs: https://deploy-preview-397--dhis2-ui-core.netlify.com
- demo: https://deploy-preview-397--dhis2-ui-core.netlify.com/demo

**Comment by varl** [comment]
`props` tables are generated by putting e.g. the string `REPLACE_WITH_PROPS_TABLE ` somewhere in the `button.md` file and then running:

```sh
npx props-table src/Button/index.js --replace REPLACE_WITH_PROPS_TABLE --output docs/components/button.md
```

The `api.md` file is simpler and can be overwritten freely:

```sh
npx props-table src/**/*.js --output docs/api.md
```

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@
 # production
 /build
 /dist
+/dest
```
Comment (Mohammer5) [inline .gitignore]: The name is a bit off, maybe `stories` suits this one better?

Comment (varl) [reply .gitignore]: Agreed, I'll change it to something else. It's not `stories` though, the stories are under `dest/demo`.

**Comment by varl** [comment]
Now with more automation to pull out the api docs.

**Comment by amcgee** [comment]
Nice!  Looks like there are still some manually-compiled docs (under `docs/components/`) - is the idea to keep those?  If so they should be listed in the sidebar (`docs/_sidebar.md`), if not they should probably be removed for now?

**Comment by varl** [comment]
They'll be removed for now. I'm working on adding jsdocs for all components that we have.

**Comment by varl** [comment]
Aight, I've added basic jsdocs for all components, will merge this into `next` so we can add more documentation strings in a self-contained manner.

**Pull request merged**